packages/headless/src/features/search/search-slice.test.ts

```diff
@@ -156,6 +156,36 @@ describe('search-slice', () => {
     expect(finalState.error).toBeNull();
   });
 
+  describe('should dispatch a logQueryError action', () => {
+    let e: MockEngine<SearchAppState>;
+    beforeEach(() => {
+      e = buildMockSearchAppEngine({state: createMockState()});
+      PlatformClient.call = jest.fn().mockImplementation(() =>
+        Promise.resolve({
+          body: {message: 'message', statusCode: 500, type: 'type'},
+        })
+      );
+    });
+
+    it('on a executeSearch error', async () => {
+      await e.dispatch(executeSearch(logSearchboxSubmit()));
+      expect(e.actions).toContainEqual(
+        expect.objectContaining({
+          type: 'search/queryError/pending',
+        })
+      );
+    });
+
+    it('on a fetchMoreResults error', async () => {
+      await e.dispatch(fetchMoreResults());
+      expect(e.actions).toContainEqual(
+        expect.objectContaining({
+          type: 'search/queryError/pending',
```

Comment:
Could we reference the string instead of hard-coding here and above? e.g. `logError().type`